packages/runtime/container-runtime/src/deltaManagerProxies.ts

```diff
@@ -99,7 +99,15 @@ export abstract class BaseDeltaManagerProxy
 		return this.deltaManager.readOnlyInfo;
 	}
 
-	constructor(protected readonly deltaManager: IDeltaManagerFull) {
+	constructor(
+		protected readonly deltaManager: IDeltaManagerFull,
+		overrides?: {
+			onReadonly?: (
+				readonly: boolean,
+				readonlyConnectionReason?: { reason: string; error?: IErrorBase },
+			) => void;
```

Comment:
Or this
```suggestion
			onReadonly?: BaseDeltaManagerProxy["onReadonly"];
```

Reply:
Actually, why not just make `onReadonly` to be  `protected` and then override it like the other proxy does for `onOp`?

Reply:
And make all those methods `protected` since this will be a common pattern with other proxies (and common pitfall...)

Reply:
it doesn't work. my guess is the object initialization flow misses the protected override during the instantiation of the base class, which is where the base class registers the event handler.